File: 백준/Gold/1253. 좋다/좋다.py
# 1253 좋다

import sys
input = sys.stdin.readline

N = int(input())
nums = list(map(int, input().strip().split()))
nums.sort()
res = 0

for i in range(N):
    target = nums[i]
    l, r = 0, N-1

    while l < r:
        if l == i:
            l += 1
            continue
        if r == i:
            r -= 1
            continue

        now = nums[l]+nums[r]

        if now == target:
            res += 1
            break
        elif now < target:
            l += 1
        else:
            r -= 1

print(res)

File: 백준/Gold/1253. 좋다/좋다.cc
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;
// boj 1253번 좋다 

int N;
int l, r;
long now;
int main() {
	cin >> N;
	int goodNums = 0;
	vector<int> arr(N);
	for (int i = 0; i < N; i++) {
		cin >> arr[i];
	}
	sort(arr.begin(), arr.end());
	
	for (int i = 0; i < N; i++) {
		int target = arr[i];
		l = 0;
		r = N - 1;

		while (l < r) {
			if (l == i) {
				l += 1;
				continue;
			}
			if (r == i) {
				r -= 1;
				continue;
			}

			now = arr[l] + arr[r];
			if (now == target) {
				goodNums++;
				break;
			}
			else if (now < target) {
				l += 1;
			}
			else {
				r -= 1;
			}
		}

	}
	cout << goodNums;
	return 0;
}